Argo/Types/Decoded/Alternative.swift

```diff
@@ -13,8 +13,15 @@
 infix operator <|> { associativity left precedence 140 }
 
 public func <|> <T>(lhs: Decoded<T>, rhs: Decoded<T>) -> Decoded<T> {
-  if case .Success = lhs {
-    return lhs
+  return lhs.alternative(rhs)
+}
+
+public extension Decoded {
+  func alternative<T>(other: Decoded<T>) -> Decoded<T> {
+    if case .Success = self {
```

Comment:
Can we change this to using a `switch`?

```swift
switch self {
case .Success: return self
case .Failure: return other
}
```

You shouldn't need any force casting, and I think it's a bit more explicit about the intent.